pkgs/development/tools/ilspy/default.nix

```diff
@@ -0,0 +1,45 @@
+{ pkgs ? (import <nixpkgs> {})}:
+with pkgs;
+let 
+  libs = [
+    xorg.libX11
+    stdenv.cc.cc.lib
+    kerberos
+    curl
+    lttng-ust
+    gtk3
+    openssl.out
+    icu
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "ilspy";
+  version = "5.0-rc2";
+
+  nativeBuildInputs = [ 
+    autoPatchelfHook
+  ];
+
+  dontStrip = true;
+  sourceRoot = ".";
+
+  buildInputs = [
+    makeWrapper
+    unzip
+  ] ++ libs;
+
+  src = fetchurl {
+    url = "https://github.com/icsharpcode/AvaloniaILSpy/releases/download/v${version}/ILSpy-linux-x64-Release.zip";
+    sha256 = "1p1fkdkpbcqj2q4n7y241lzx98w33vmh4n2gahw7lbrrg1p7rxxa";
+  };
+
+  installPhase = ''
```

Comment:
Hi, and thanks for reviewing! I didn't build from source because I don't need it, it's more work, and I don't even have any other platforms to test things.

Reply:
For `nixpkgs` generally build from source, since it is then linked correctly (avoids the `LD_LIBRARY_PATH` environment variable) and we can support platforms as they are added to NixOS. Additionally we can catch more NixOS-related issues of the package while building which often only show themselves when running in the case of prebuilt packages.

Overall the resulting packages are also cleaner and more transparent as to what is going on.

I understand it is more work and it often can be kind of tricky to get more obscure build systems to work with nix. If you don't want to rework it for NixOS, you can still use `callPackage` from your `configuration.nix` to use it on your system or [override your nixpkgs](https://nixos.org/nixpkgs/manual/#chap-overrides).

Reply:
Unfortunately I don't think I'll spend any more time on this any time soon. It does work so I think it should be merged (i.e. I think it's better than not even having a package for this), but if it might cause trouble for maintainers I understand why maintainers wouldn't want to merge it.

Reply:
Fair enough, as said you can always use overriding to add local packages to nixpkgs for your system.